### PR TITLE
Feature: make data dir configurable

### PR DESCRIPTION
**Pull request by ronaldyuwandika**
This is PR for fix this issue #3 
 
What I do:

- add support to dynamically change the default CSI driver rawfile and data directory

How to tests:

- Just change the value of  `rawfile_dir` and `data_dir` in the helm values file stored at `deploy/charts/rawfile-csi/values.yaml`

Signed-off-by: ronaldyuwandika ronald.d.yuwandika@gmail.com

### CODE REVIEW

**Comment by jshin47** [comment]
The CI is failing but I don't think it has anything to do with your code...

**Comment by ronaldyuwandika** [comment]
I think the CI is failing because there's a problem with the docker build and push stage that needs username and password for dockerhub repository

File: deploy/charts/rawfile-csi/values.yaml [hunk 1]
```diff
@@ -12,6 +12,10 @@ defaults: &defaults
     requests:
       cpu: 10m
       memory: 100Mi
+  rawfile_dir: /rawfile
```
Comment (SeMeKh) [inline deploy/charts/rawfile-csi/values.yaml:15]: Please change this to `/var/csi/rawfile` to keep it backward-compatible.

File: Dockerfile [hunk 2]
```diff
@@ -19,5 +19,7 @@ ENV PYTHONUNBUFFERED 1
 
 ARG IMAGE_TAG
 ARG IMAGE_REPOSITORY
+ARG DATA_DIR
```
Comment (SeMeKh) [inline Dockerfile:22]: Let's not make data_dir configurable. This is what the container sees, and does not need to be changed.

**Review by SeMeKh — CHANGES_REQUESTED** [review]
Thanks for your contribution. Will you please fix the comments?

File: deploy/charts/rawfile-csi/values.yaml (hunk 1 above)
Comment (ronaldyuwandika) [reply deploy/charts/rawfile-csi/values.yaml:15]: sure, I will change it later

File: Dockerfile (hunk 2 above)
Comment (ronaldyuwandika) [reply Dockerfile:22]: Ah okay, I'll undo this change

File: deploy/charts/rawfile-csi/values.yaml (hunk 1 above)
Comment (ronaldyuwandika) [reply deploy/charts/rawfile-csi/values.yaml]: hello, @SeMeKh I've updated this to `/var/csi/rawfile`
can you please check it again?
thank you

File: deploy/charts/rawfile-csi/templates/01-node-plugin.yaml [hunk 3]
```diff
@@ -66,6 +66,8 @@ spec:
               value: unix:///csi/csi.sock
             - name: IMAGE_REPOSITORY
               value: "{{ .Values.node.image.repository }}"
+            - name: DATA_DIR
```
Comment (SeMeKh) [inline deploy/charts/rawfile-csi/templates/01-node-plugin.yaml:69]: The "remove data_dir variable" applies to this one too

File: deploy/charts/rawfile-csi/templates/01-node-plugin.yaml [hunk 4]
```diff
@@ -85,7 +87,7 @@ spec:
               mountPath: /var/lib/kubelet
               mountPropagation: "Bidirectional"
             - name: data-dir
-              mountPath: /data
+              mountPath: {{ .Values.node.data_dir }}
```
Comment (SeMeKh) [inline deploy/charts/rawfile-csi/templates/01-node-plugin.yaml:90]: ditto

File: deploy/charts/rawfile-csi/values.yaml [hunk 5]
```diff
@@ -12,6 +12,10 @@ defaults: &defaults
     requests:
       cpu: 10m
       memory: 100Mi
+  rawfile_dir: /var/csi/rawfile
+  data_dir: /data
```
Comment (SeMeKh) [inline deploy/charts/rawfile-csi/values.yaml:16]: ditto

File: templates/task.yaml [hunk 6]
```diff
@@ -20,7 +20,7 @@ spec:
       imagePullPolicy: IfNotPresent
       volumeMounts:
         - name: data-dir
-          mountPath: /data
+          mountPath: {data_dir}
```
Comment (SeMeKh) [inline templates/task.yaml:23]: ditto

**Review by SeMeKh — APPROVED** [review]
Thanks. Fix these and it's good to go.

File: templates/task.yaml [hunk 7]
```diff
@@ -11,7 +11,7 @@ spec:
   volumes:
     - name: data-dir
       hostPath:
-        path: /var/csi/rawfile
+        path: {rawfile_dir}
```
Comment (SeMeKh) [inline templates/task.yaml:14]: Actually, this needs to be passed into the templating function so that it can get rendered.

**Review by SeMeKh — CHANGES_REQUESTED** [review]

File: templates/task.yaml (hunk 7 above)
Comment (ronaldyuwandika) [reply templates/task.yaml:14]: image
can i change it into templating function like this?

Comment (SeMeKh) [reply templates/task.yaml:14]: The values are interpolated not by helm, but the python code. So that's where the code needs to be updated.

**Pull request closed without merging**